### PR TITLE
REFACTOR-#7002: get more information about exceptions from `eval_general` utility

### LINKED ISSUE #7002: Refactor `eval_general` utility to get more information about exceptions

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #7002 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/utils.py [hunk 1]
```diff
@@ -863,6 +863,10 @@ def check_df_columns_have_nans(df, cols):
     )
 
 
+class TestError(Exception):
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:866]: Could not find a suitable built-in exception type.

File: modin/pandas/test/utils.py [hunk 2]
```diff
@@ -888,24 +892,29 @@ def execute_callable(fn, inplace=False, md_kwargs={}, pd_kwargs={}):
...
-                assert isinstance(
-                    md_e.value, type(pd_e)
-                ), "Got Modin Exception type {}, but pandas Exception type {} was expected".format(
-                    type(md_e.value), type(pd_e)
+            except Exception as md_e:
+                if check_exception_type:
+                    assert isinstance(
+                        md_e, type(pd_e)
+                    ), "Got Modin Exception type {}, but pandas Exception type {} was expected".format(
+                        type(md_e), type(pd_e)
```
Comment (YarShev) [inline modin/pandas/test/utils.py:908]: Previously, we used md_e.value. Why did we discard it?

File: modin/pandas/test/utils.py (hunk 1 above)
Comment (YarShev) [inline modin/pandas/test/utils.py:866]: Maybe `NoModinException`?

File: modin/pandas/test/utils.py (hunk 2 above)
Comment (anmyachev) [reply modin/pandas/test/utils.py:908]: This is an implementation detail of how pytest catches exceptions.  Now we rely on python itself.

File: modin/pandas/test/utils.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/utils.py]: ok

**Review by dchigarev — APPROVED** [review]

**Pull request merged**